File: Java/SelectionSort/src/domain/SelectionSort.java
package domain;

public class SelectionSort {
	
	public static void byAuthor(Book[] _books) {
		for(int i = 0; i < _books.length; i++) {
			int minIndex = i;
			for(int j = i+1; j < _books.length; j++) {
				
				if(_books[j].compareAuthor(_books[minIndex]) < 0) {
					minIndex = j;
				}
			}
			if (minIndex != i) {
				swapBooks(_books, i, minIndex);
			}
		}
	}
	
	public static void byNumberPages(Book[] _books) {
		for(int i = 0; i < _books.length; i++) {
			int minIndex = i;
			for(int j = i+1; j < _books.length; j++) {
				if(_books[j].compareNumberPages(_books[minIndex]) < 0) {
					minIndex = j;
				}
			}
			if (minIndex != i) {
				swapBooks(_books, i, minIndex);
			}
		}
	}
	
	public static void byTitle(Book[] _books) {
		for(int i = 0; i < _books.length; i++) {
			int minIndex = i;
			for(int j = i+1; j < _books.length; j++) {
				if(_books[j].compareTitle(_books[minIndex]) < 0) {
					minIndex = j;
				}
			}
			if (minIndex != i) {
				swapBooks(_books, i, minIndex);
			}
		}
	}
	
	private static void swapBooks(Book[] _books, int swap1, int swap2) {
		Book temp = _books[swap1];
		_books[swap1] = _books[swap2];
		_books[swap2] = temp;
	}
}

File: Python/src/SelectionSort.py
class SelectionSort:

    @staticmethod
    def byAuthor(books):
        booksLength = len(books)
        for index in range(booksLength):
            minIndex = index
            for j in range(index+1, booksLength):
                if books[j].compareAuthor(books[minIndex]) < 0:
                    minIndex = j
            if minIndex != index:
                SelectionSort.swapBooks(books, index, minIndex)

    @staticmethod
    def byNumberPages(books):
        booksLength = len(books)
        for index in range(booksLength):
            minIndex = index
            for j in range(index+1, booksLength):
                if books[j].compareNumberPages(books[minIndex]) < 0:
                    minIndex = j
            if minIndex != index:
                SelectionSort.swapBooks(books, index, minIndex)

    @staticmethod
    def byTitle(books):
        booksLength = len(books)
        for index in range(booksLength):
            minIndex = index
            for j in range(index+1, booksLength):
                if books[j].compareTitle(books[minIndex]) < 0:
                    minIndex = j
            if minIndex != index:
                SelectionSort.swapBooks(books, index, minIndex)

    @staticmethod
    def swapBooks(books, swap1, swap2):
        temp = books[swap1]
        books[swap1] = books[swap2]
        books[swap2] = temp
